sphinx/domains/cpp.py

```diff
@@ -4409,8 +4408,12 @@ class DefinitionParser(BaseParser):
 
     _prefix_keys = ('class', 'struct', 'enum', 'union', 'typename')
 
-    def __init__(self, definition: Any, warnEnv: Any, config: "Config") -> None:
-        super().__init__(definition, warnEnv)
+    def __init__(self, definition: str, *,
+                 location: Union[nodes.Node, Tuple[str, int]],
+                 emitWarnings: bool,
```

Comment:
I'll try to suppress `emitWarning` parameter as a refactoring.

Reply:
I just made a simple helper: #7327

Reply:
Awesome! This simplifies things a lot. Thanks!